### ISSUE TITLE
Ability to set enums in config

### ISSUE THREAD

**Issue by SHaaD94** [carueda/tscfg#62]
It would be cool to have an ability to set a field which might have only limited set of values for example, let's say **fruitType** can be only **apple, banana or pineapple**.
In the config it could look like this:
```
app.fruitType = "string E ["apple", "banana", "pineapple"]"
```
After it, sealed trait with three classes could be generated:
```
trait FruitType 
object FruitType {
   object Apple extends FruitType
   object Banana extends FruitType
   object Pineapple extends FruitType
}
```

**Comment by carueda** [comment]
Adding enums would be great, thanks.

At the spec level, maybe something combined with or similar to [`@define`](https://github.com/carueda/tscfg#shared-object) (#54):

```
#@define enum
FruitType = [apple, banana, pineapple]

# then, somewhere for the actual use/reference:

foo: {
  fruit: FruitType,
  ...
}
```

**Comment by johanneshiry** [comment]
hi @SHaaD94 - are you interested in actively working to provide this feature? I handed in #64 which would make a huge benefit in several of our projects and as @carueda suggested I might make sense to talke your proposal and #64 together ...

**Comment by carueda** [comment]
@SHaaD94 / @johanneshiry :  fyi, i've done some progress on this, but will have to postpone completion for a little while (the java generation seems pretty complete, but not yet the scala one, at least for enum references at the root of the given config). I realize that enhancements like this and #64 are not as "straightforward" as one would like them to be given that the code has become rather complex; I myself have to spend some time catching up with the structure of things, etc. (I wish I had more time to proceed with a general revision.)  Thanks again for your suggestions and contributions, really appreciated.

**Comment by carueda** [comment]
Implemented in 0.9.983 - https://github.com/carueda/tscfg#enum
